### ISSUE TITLE
Interaction of Arcane Missiles with Arcane Potency talent - blizzlike or not?

### ISSUE THREAD

**Issue by banana-bank** [chromiecraft/chromiecraft#1308]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [X] 30-39
- [ ] 40-49
- [ ] 50-59

### Current Behaviour

Casting Arcane Missiles with Arcane Potency procced, either by activating presence of mind or through a proc, each discrete missile interacts with Arcane Potency the same way other damage spells do (consumes the buff when a hit lands on the target)

It is possible to gain increased crit for more than 1 individual missile, but only if you are far enough from the target that the first missile has not hit the target by the time the second missile leaves the caster's hands.

### Expected Blizzlike Behaviour

Arcane Potency seemingly not interact with Arcane Missiles the way it interacts with every other damaging non channeled mage spell. It used to modify crit for the duration of an Arcane Missiles channel, but was immediately consumed upon proccing either when the player begins casting arcane missiles or if they are already channeling it - in effect, Arcane Missiles had its own special interaction with Arcane Potency, where the spell channel start/end, and not the projectiles hitting, interacts with Arcane Potency.

This ensured that each missile fired benefits from the boosted crit chance. it also ensures it is impossible to get boosted crit for more than one spellcast - eg. if potency simply did not get consumed by arcane missiles, as it was like in the beta, you could wait until arcane missiles has 1 tick left, then immediately fire a PoM pyro, giving 30% crit for the entire AM as well as the pyro, but instead with the hidden interaction the crit% is immediately removed as soon as the player stops channeling AM. 

### Source

https://youtu.be/eLe1GacBNsk?t=530
Play at 0.25 speed from linked timestamp

Observe the combat text/buff icon when player is casting AM with ArcPot procced, the buff disappears as soon as it appears (inbetween missile hits) and most spells in the channel does crit. So the proc is not being consumed by a projectile. Since player memories all point towards boosted crit for entire channel is it tough to find concrete video evidence of this but this does point towards a different implementation than the existing one with missiles consuming potency as they strike the target.

https://youtu.be/WYORj0rNWhY?t=97
Frameskip thru arcane missiles cast at 1:37 - potency buff procs mid-channel and is consumed between missile hits rather than when the next missile hits.

https://wowpedia.fandom.com/wiki/Arcane_Potency
- Makes references to legacy behavior and the change of Arcane Missiles channel consuming and applying a custom form of Arcane Potency. however this may apply to when clearcasting could only proc at the start of channeling an Arcane Missiles. Again the core question is if this is different in 3.3.5 or not after clearcasting was changed to be able to proc mid-arcane missiles channel... And its almost impossible to find evidence of this in legacy videos because it is a 30% and not 100% buff.

BUT since the alternative is that arcane missiles simply "swallows" the crit buff and provides 0% crit for one of the bread and butter spells of arcane, and because player memories point towards it working the other way around, I personally believe that Potency always "snapshot" the +30% crit until you finished channeling Arcane Missiles, especially because this is how it worked throughout most of WOTLK patch cycle and a nerf would have generated more historical posts about the topic.

### Steps to reproduce the problem

1. Cast Arcane Missiles on target dummy with both talents, observe when the Arcane Potency buff appears and disappears

2. Cast PoM and then cast Arcane Missiles at max range from target, pref. with the range boosting talent, the buff will be immediately consumed when the first missile hits, but remain active before any missiles hit the dummy, any generated missile before the target is hit will benefit from the boosted crit chance but the crit chance of the entire arcane missiles channel will not benefit from Arcane Potency

### Extra Notes

I am partly making this report to see a discussion about the topic too since I see a bit of "mandela effect" among veteran Mage players and other original WOTLK player accounts about the exact working of this mechanic, and how it pertains to 3.3.5 implementation.

Unclear about interaction with the Clearcasting change in 3.3.5 that causes individual Arcane Missiles projectiles to roll for a reduced chance of clearcasting. I assume that if a channel of Arcane Missiles causes Clearcasting and Arcane Potency gain towards the tail end of the channel, the buff is consumed and only that cast of Arcane Missiles yields the crit boost. 

I believe the references to "double tapping" procs found on wowwiki and old discussion are from before the clearcasting change. But even with this change the hardcoded interaction between ArcPot and Arcane Missiles was not changed. The problem is gathering evidence/figuring out if this also changed in 3.3.5 or if it stayed the same.

https://wowpedia.fandom.com/wiki/Talk:Arcane_Potency

For instance in this historic discussion you can see proof that at least before 3.X change to arcane missiles and potency interaction, the crit was online for the entire channel, but all we know about the implementation here is that it is somehow different from blizz but not in what exact fashion.. frustratingly enough.

I also found some more technical discussion here:
https://github.com/TrinityCore/TrinityCore/issues/18813

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/3d4befd7043fbbceb32f7c2a6e0a4f9f638f8b0a

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-Zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by Azcobu** [comment]
Thanks for the report and the research that went into it. There does seem to be evidence that the Arcane Potency buff's interaction with Arcane Missiles is not as straightforward as it is on Chromie, specifically your point about the buff not being consumed by a single Arcane Missile projectile. The same behaviour is shown here:

https://youtu.be/3NLSkJRDaso?t=474

If you step through this frame by frame, you can see that Clearcasting procs and Arcane Potency with it, and then Arcane Potency disappears just a frame or two before the first projectile even hits the target. The two YT videos you linked also showed that behaviour. The problem is of course, we can't see the crit chance for the rest of the volley, so we can't see if the crit bonus has been snapshotted for the rest of the channel. The problem is that although it does sound plausible, it would be a pretty significant change in the way this spell works, and so far, I haven't yet found any unambiguous source that confirms snapshotting.

Some more notes here: 
https://github.com/azerothcore/azerothcore-wotlk/issues/4563
https://github.com/ccshiro/cc-buglist/issues/1571

**Comment by banana-bank** [comment]
Aye, it's tough to find 100% cold hard evidence. Before making the post I asked around and the replies I got were "yes, it snapshots, always did", some said "it only snapshots if you are close enough" (which i dont understand) and some just said "it has always been broken on pservers". And me personally I was levelling a mage and decided to switch to Arcane after I could spec 2/2 potency but then was surprised it did not work the way I remembered. 

I'm still looking around to see if I can find more info.

**Comment by Salehyn** [comment]
Ok I've done some digging and my results say this is working as intended.

More in detail: Each Arcane Missile tick has an individual 2%ish chance to proc Clearcast (As of 3.3.0), and therefore Arcane Potency. This 30% crit boost buff is consumed when the next Arcane Missile tick hits the target (as shown on videos), which takes a good 2-3seconds if the mage stands at maximum range. As long as the Missile is traveling, the buff is still active and affects any missile that is fired before the impact. This means easily 3-4 ticks with Missile Barrage firing them every half second.
Any missiles that fire from the spell after the first missile landed and the buff was consumed is no longer affected by the crit boost.
This is how it currently works on CC and also how it seems to work on wotlk classic as well.

Let me demonstrate this with logs.

On most encounters where the mages are positioned at the usual max range, the crit rate of Arcane Missiles is comparable to Arcane Blasts, being just a few % lower, a discrepancy that could be due to RNG, controlled Presence of Mind cds used on Blasts, or pretty much anything else.
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=21&sourcebuffs=-57531 (algalon)
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=43 (mimiron)
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=13 (xt)
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=23 (kologarn)
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=50 (vezax)
(To rule out outside factors, I chose parses for specifically fights that involve no temporary crit boosts from boss mechanics.)

This clearly shows more than 1 tick of each Missile is affected by the Arcane Potency crit boost. (note: mages around togc gear level have only around 55-60% crit chance fully raid buffed, and there is a notable increase to that via the procs, comparable to the boost Arcane Blast receives.)
If the crit from Arcane Potency applied to merely one tick of each missile cast, these numbers simply wouldn't be possible. 
However, Arcane Blast does have a small edge against Missiles in crit rates, which points towards Missile ticks not having the exact same statistical crit chances.

Now, let's see logs for encounters where mages are typically stacked in close range:

https://classic.warcraftlogs.com/reports/TN9rCxmHFLv8dMak#fight=7&type=damage-done&source=31 (valkyrs)
https://classic.warcraftlogs.com/reports/1D6WGZc7aY4FhC9j#fight=8&type=damage-done&source=2 (valkyrs)
https://classic.warcraftlogs.com/reports/GbHWFkcrgQqyVTjp#fight=6&type=damage-done&source=2 (valkyrs)
https://classic.warcraftlogs.com/reports/y2gVmnCkJjzPHapB#type=damage-done&source=36&fight=10 (ignis)
https://classic.warcraftlogs.com/reports/zDwdrKGkQtnq2fpC#boss=-2&wipes=2&start=3991486&end=8840750&type=damage-done&source=19 (ignis)

Suddenly you see a huge discrepancy (10-15%) in crit rates between Blasts and Missiles. Missiles are down to 60-64% crit, which is barely higher than the mage's default raid buffed crit chance (approx 55-60% at these gear levels). This is purely due to the mage's positioning, as these encounters favor stacking melee and ranged for healing and mechanics, which means that every time the mage uses Arcane Missiles with the Potency buff active, it will be consumed by the first missile landing immediately, rather than having a travel time that allows the buff to affect subsequent missiles.

Summary: numbers on logs consistently show that Arcane Missiles have a shortfall on crit rates when compared to Arcane Blasts. This gap is insignificant on encounters that allow the mage to stand at max range and benefit multiple missile ticks from Arcane Potency procs, while it is very significant on fights where the raid is stacked in melee range.

-----------------------
Bottom line: this spell interaction is working as intended. Mages need to stand at maximum range and ensure missile barrage procs to maximize the benefit from Arcane Potency when using Arcane Missiles.
